# Hypothesis 2

In [135]:
# Importing the necessary libraries
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as st
import numpy as np
from statsmodels.formula.api import ols
import statsmodels.api as sm
import HT as ht

Using ANOVA as another method of hypothesis testing

In [118]:
data = pd.read_csv('final_df.csv', index_col = 0)
data.head()

Player    FG%    3P%    2P%    FT%  ORB/G  DRB/G  TRB/G  AST/G  \
0    A.C. Green  0.442  0.062  0.448  0.675  1.890  3.722  5.615  0.865   
1   A.J. Guyton  0.384  0.382  0.381  0.824  0.285  0.750  1.035  1.870   
2  A.J. Hammons  0.405  0.500  0.375  0.450  0.360  1.270  1.640  0.180   
3    A.J. Price  0.373  0.273  0.447  0.632  0.251  1.093  1.344  1.971   
4  Aaron Brooks  0.410  0.359  0.444  0.838  0.403  1.268  1.671  3.174   

   STL/G  BLK/G  TOV/G   PF/G  PTS/G POS  height  weight  
0  0.633  0.202  0.602  1.572  5.440  PF   203.0   106.0  
1  0.245  0.155  0.775  0.785  5.710  PG   208.0    99.0  
2  0.050  0.590  0.450  0.950  2.180   C   198.0    99.0  
3  0.401  0.021  0.753  0.735  5.578  PG   213.0   124.0  
4  0.591  0.134  1.582  1.932  9.604  PG   183.0    73.0

Using ANOVA, we will be subsetting the data based on position, and use blocks per game as the dependent variable. With ANOVA, we can use a statistcal test which analyses the overall variance among the different positions 

In [6]:
data.POS.value_counts()
# As there are 12 players that have played in two positions, I will get rid of these players as they represent
# a small proportion of the total players (< 0.01%)

PG       358
SG       349
SF       310
PF       297
C        232
SG-SF      4
C-PF       4
SG-PG      2
PG-SG      1
SF-PF      1
Name: POS, dtype: int64

In [ ]:
data = data.loc[:]

In [30]:
PG_subset = data.loc[data['POS'] == 'PG']
display(PG_subset.head(2))
SG_subset = data.loc[data['POS'] == 'SG']
display(SG_subset.head(2))
SF_subset = data.loc[data['POS'] == 'SF']
display(SF_subset.head(2))
PF_subset = data.loc[data['POS'] == 'PF']
display(PF_subset.head(2))
C_subset = data.loc[data['POS'] == 'C']
display(C_subset.head(2))

Player    FG%    3P%    2P%    FT%  ORB/G  DRB/G  TRB/G  AST/G  STL/G  \
1  A.J. Guyton  0.384  0.382  0.381  0.824  0.285  0.750  1.035  1.870  0.245   
3   A.J. Price  0.373  0.273  0.447  0.632  0.251  1.093  1.344  1.971  0.401   

   BLK/G  TOV/G   PF/G  PTS/G POS  height  weight  
1  0.155  0.775  0.785  5.710  PG   208.0    99.0  
3  0.021  0.753  0.735  5.578  PG   213.0   124.0

Player    FG%    3P%    2P%    FT%  ORB/G  DRB/G  TRB/G  AST/G  \
7    Aaron Harrison  0.132  0.150  0.111  0.458  0.095   0.56  0.655  0.350   
11  Adam Harrington  0.294  0.351  0.269  0.750  0.073   0.45  0.523  0.817   

    STL/G  BLK/G  TOV/G   PF/G  PTS/G POS  height  weight  
7   0.145  0.000  0.095  0.440  0.530  SG   198.0    95.0  
11  0.120  0.043  0.087  0.523  1.867  SG   196.0    90.0

Player    FG%    3P%    2P%    FT%  ORB/G  DRB/G  TRB/G  AST/G  \
16    Adonis Thomas  0.381  0.100  0.528  1.000  0.000  0.625  0.625  0.500   
23  Al-Farouq Aminu  0.425  0.291  0.464  0.726  1.401  4.309  5.710  1.231   

    STL/G  BLK/G  TOV/G   PF/G  PTS/G POS  height  weight  
16  0.000  0.000  0.085  0.750  2.040  SF   201.0    90.0  
23  0.951  0.599  1.289  1.859  7.237  SF   206.0    99.0

Player    FG%    3P%    2P%    FT%  ORB/G  DRB/G  TRB/G  AST/G  \
0    A.C. Green  0.442  0.062  0.448  0.675  1.890  3.722  5.615  0.865   
5  Aaron Gordon  0.458  0.285  0.520  0.703  1.467  3.587  5.053  1.407   

   STL/G  BLK/G  TOV/G   PF/G  PTS/G POS  height  weight  
0  0.633  0.202  0.602  1.572  5.440  PF   203.0   106.0  
5  0.670  0.560  0.923  1.960  9.043  PF   206.0    99.0

Player    FG%  3P%    2P%    FT%  ORB/G  DRB/G  TRB/G  AST/G  STL/G  \
2  A.J. Hammons  0.405  0.5  0.375  0.450  0.360   1.27   1.64  0.180   0.05   
6    Aaron Gray  0.466  0.0  0.471  0.562  1.242   1.96   3.20  0.665   0.29   

   BLK/G  TOV/G   PF/G  PTS/G POS  height  weight  
2   0.59  0.450  0.950  2.180   C   198.0    99.0  
6   0.26  0.825  1.948  2.845   C   213.0   122.0

In [68]:
column_names = ['POS', 'BLK/G']
def POS_BLK_df(data):
    new_df = data.loc[:,column_names]
    return new_df

In [121]:
PG_BLK = POS_BLK_df(PG_subset)
SG_BLK = POS_BLK_df(SG_subset)
SF_BLK = POS_BLK_df(SF_subset)
PF_BLK = POS_BLK_df(PF_subset)
C_BLK = POS_BLK_df(C_subset)

In [131]:
list_of_dataframes = [PG_BLK, SG_BLK, SF_BLK, PF_BLK, C_BLK]
anova_test = pd.concat(list_of_dataframes)
print(anova_test.shape)
anova_test.reset_index(inplace = True)
anova_test.drop('index', axis = 1, inplace = True)
anova_test.head()

(1546, 2)


POS  BLK/G
0  PG  0.155
1  PG  0.021
2  PG  0.134
3  PG  0.022
4  PG  0.000

In [132]:
formula = "BLK ~ C(POS)"
anova_test['BLK'] = anova_test['BLK/G']
lm = ols(formula, anova_test).fit()
table = sm.stats.anova_lm(lm, typ = 2)
print(table)

              sum_sq      df           F         PR(>F)
C(POS)     79.314438     4.0  196.649897  2.608512e-136
Residual  155.382167  1541.0         NaN            NaN


Can see a very significant effect of position on blocks per game. Our F-statistic is 196
For a more meaningful analysis we can use Welch's t-test in pairwise comparisons between subsets.
We may discover more drastic differences in t-statistics between certain groups 

# Pairwise Comparisons 

## Comparing Point Guards and Centers

In [137]:
display(PG_BLK.head(), C_BLK.head())

POS  BLK/G
1   PG  0.155
3   PG  0.021
4   PG  0.134
10  PG  0.022
15  PG  0.000

POS  BLK/G
2    C  0.590
6    C  0.260
14   C  1.725
20   C  1.241
26   C  0.680

In [ ]:
sampling = CLT()
ht.sampling.